apps/services/user-profile/src/app/v2/islykill.service.ts

```diff
@@ -15,20 +53,64 @@ export class IslykillService {
     email?: string
     phoneNumber?: string
   }): Promise<PublicUser> {
-    return this.islyklarApi
-      .islyklarPut({
-        user: {
-          ssn: nationalId,
-          ...(email && { email }),
-          ...(phoneNumber && { mobile: phoneNumber }),
-        },
+    try {
+      const user = {
+        ssn: nationalId,
+        ...(email && { email }),
+        ...(phoneNumber && { mobile: phoneNumber }),
+      }
+
+      return await this.islyklarApi.islyklarPut({ user })
```

Comment:
You have redundant return await in some places. You should be good by just returning